app/src/main/java/com/example/candyshop/ui/DetailsScreenViewModel.kt

```diff
@@ -3,37 +3,69 @@ package com.example.candyshop.ui
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.setValue
+import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
-import androidx.lifecycle.ViewModelProvider
-import androidx.lifecycle.ViewModelProvider.AndroidViewModelFactory.Companion.APPLICATION_KEY
-import androidx.lifecycle.viewmodel.initializer
-import androidx.lifecycle.viewmodel.viewModelFactory
-import com.example.candyshop.CandyApplication
-import com.example.candyshop.database.TempDatabase
-import com.example.candyshop.network.CandyItem
-
-class DetailsScreenViewModel(private val temporaryDb: TempDatabase) : ViewModel() {
+import androidx.lifecycle.viewModelScope
+import com.example.candyshop.api.CandyItemsRepository
+import com.example.candyshop.data.Resource
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class DetailsScreenViewModel @Inject constructor(
+    private val candyItemsRepository: CandyItemsRepository,
+    private val savedStateHandle: SavedStateHandle
+) : ViewModel() {
     var textFieldInput by mutableStateOf("")
```

Comment:
this textFieldInput and textFieldInput maybe could be also part of DetailsState. Also try to not expose outside the class any mutable states